azurelinuxagent/ga/exthandlers.py

```diff
@@ -206,14 +205,13 @@ class ExtHandlerState(object):
     FailedUpgrade = "FailedUpgrade"
 
 
-def get_exthandlers_handler():
-    return ExtHandlersHandler()
+def get_exthandlers_handler(protocol):
```

Comment:
Now the protocol is instantiated in the main loop and passed to the extension handler as parameter